plugins/PRAplugin/src/ETmodel.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import division, print_function , unicode_literals, absolute_import
+import warnings
+warnings.simplefilter('default', DeprecationWarning)
+
+#External Modules---------------------------------------------------------------
+import numpy as np
+import math
+#External Modules End-----------------------------------------------------------
+
+#Internal Modules---------------------------------------------------------------
+from PluginsBaseClasses.ExternalModelPluginBase import ExternalModelPluginBase
+from PostProcessors.ETstructure import ETstructure
+#Internal Modules End-----------------------------------------------------------
+
+
+class ETmodel(ExternalModelPluginBase):
+  """
+    This class is designed to create an Event-Tree model
+  """
+
+  def _readMoreXML(self, container, xmlNode):
+    """
+      Method to read the portion of the XML that belongs to the Event-Tree model
+      @ In, container, object, self-like object where all the variables can be stored
+      @ In, xmlNode, xml.etree.ElementTree.Element, XML node that needs to be read
+      @ Out, None
+    """
+    container.mapping    = {}
+    container.InvMapping = {}
+
+    for child in xmlNode:
+      if child.tag == 'topEvents':
+        container.topEventID = child.text.strip()
+      elif child.tag == 'map':
+        container.mapping[child.get('var')]      = child.text.strip()
+        container.InvMapping[child.text.strip()] = child.get('var')
+      elif child.tag == 'variables':
+        variables = [str(var.strip()) for var in child.text.split(",")]
+      elif child.tag == 'sequenceID':
+        container.sequenceID = child.text.strip()
+      else:
+        raise IOError("ETmodel: xml node " + str (child.tag) + " is not allowed")
```

Comment:
I think we have the message handler in the plugin

Reply:
message handler is not available.